src/Predictors/MonteCarloPredictor.cpp

```diff
@@ -151,7 +151,7 @@ namespace PCOE {
             double timeOfCurrentSavePt = std::numeric_limits<double>::infinity();
             auto currentSavePt = savePts.begin();
             if (currentSavePt != savePts.end()) {
-                timeOfCurrentSavePt = (*currentSavePt).time_since_epoch().count();
+                timeOfCurrentSavePt = (*currentSavePt).time_since_epoch().count() / 1000000;
```

Comment:
The right-hand side of this equation is an integer. Recommend using the `seconds` function (defined in MessageClock.h because I don't know where else to put it), i.e. `timeOfCurrentSavePt = seconds(*currentSavePt)`